tests/test_api_skeleton.py

```diff
@@ -464,3 +467,27 @@ def test_cran_license(package, license_id, license_family, license_files, testin
     license_files = ensure_list(license_files)
     for m_license_file in m_license_files:
         assert os.path.basename(m_license_file) in license_files
+
+
+@pytest.mark.skipif(not external.find_executable("shellcheck"), reason="requires shellcheck >=0.7.0")
+@pytest.mark.parametrize(
+    "package, repo", [("r-usethis", "cran"), ("Perl::Lint", "cpan"), ("screen", "rpm")]
+)
+def test_build_sh_shellcheck_clean(package, repo, testing_workdir, testing_config):
+    api.skeletonize(packages=package, repo=repo, output_dir=testing_workdir, config=testing_config)
+
+    build_sh = glob(os.path.join(testing_workdir, "**", "build.sh"), recursive=True)[0]
+    cmd = [
+        "shellcheck",
+        "--enable=all",
+        # SC2154: var is referenced but not assigned,
+        #         see https://github.com/koalaman/shellcheck/wiki/SC2154
+        "--exclude=SC2154",
+        build_sh,
+    ]
+
+    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Comment:
If you want to check if the command was successful you can use ``subprocess.check_call`` instead of ``Popen`` directly. ``check_call`` is just a wrapper around ``Popen`` which will raise ``CalledProcessError`` if the return exit is different of zero.

Reply:
Good point. Yes `check_call` would be enough here to directly finish the testcase in case of findings. 

The idea here is to not just fail in case of findings (exit_code not 0), but also print the findings (from stdout) so that it is easier to fix them. As the `check_call` help https://docs.python.org/3/library/subprocess.html#subprocess.check_call is telling to not pipe stdout/stderr I moved to `p = Popen` + `f = p.communicate`+ `assert f == []`...

Reply:
you could use check_output if stdout is all you care about.  I think you can redirect stderr to stdout to get both with check_output.  Not a big deal, though.  It's fine the way it is in my opinion.